go/test/endtoend/tabletgateway/healthcheck/vtgate_test.go

```diff
@@ -78,6 +74,46 @@ func verifyVtgateVariables(t *testing.T, url string) {
 	assert.True(t, isMasterTabletPresent(healthCheckConnection), "Atleast one master tablet needs to be present")
 }
 
+func TestReplicaTransactions(t *testing.T) {
+	// TODO(deepthi): this test seems to depend on previous test. Fix tearDown so that tests are independent
+	defer cluster.PanicHandler(t)
+	// Healthcheck interval on tablet is set to 1s, so sleep for 2s
+	time.Sleep(2 * time.Second)
+	ctx := context.Background()
+	masterConn, err := mysql.Connect(ctx, &vtParams)
+	require.Nil(t, err)
+	replicaConn, err := mysql.Connect(ctx, &vtParams)
+	require.Nil(t, err)
+	defer masterConn.Close()
+	defer replicaConn.Close()
+
+	// insert a row using master
+	exec(t, masterConn, "insert into customer(id, email) values(1,'email1')")
+	// select using a replica
+	_ = exec(t, replicaConn, "use @replica")
+	qr := exec(t, replicaConn, "select id, email from customer")
+	// no data expected
+	assert.Equal(t, `[[INT64(1) VARCHAR("email1")]]`, fmt.Sprintf("%v", qr.Rows), "select returned wrong result")
+	// begin transaction on replica
+	_ = exec(t, replicaConn, "begin")
+	// insert more data on master
+	exec(t, masterConn, "insert into customer(id, email) values(2,'email2')")
+	// replica doesn't see new row because it hasn't been committed
+	qr = exec(t, replicaConn, "select id, email from customer")
+	assert.Equal(t, `[[INT64(1) VARCHAR("email1")]]`, fmt.Sprintf("%v", qr.Rows), "select returned wrong result")
+	// commit on master
+	_ = exec(t, masterConn, "commit")
+	// replica still doesn't see new row because it is in a transaction
+	qr = exec(t, replicaConn, "select id, email from customer")
+	assert.Equal(t, `[[INT64(1) VARCHAR("email1")]]`, fmt.Sprintf("%v", qr.Rows), "select returned wrong result")
+	// close transaction on replica
```

Comment:
how do we end a replica transaction? commit/rollback doesn't make sense.

Reply:
That is how it is :)

Reply:
:+1: